plugins/slicer/MONAILabel/MONAILabel.py

```diff
@@ -524,6 +526,35 @@ def updateGUIFromParameterNode(self, caller=None, event=None):
         idx = 0 if idx < 0 < self.ui.labelComboBox.count else idx
         self.ui.labelComboBox.setCurrentIndex(idx)
 
+        if self._segmentNode:
+            segmentation = self._segmentNode.GetSegmentation()
+            totalSegments = segmentation.GetNumberOfSegments()
+            segmentIds = [
+                segmentation.GetNthSegmentID(i)
+                for i in range(totalSegments)
+                if segmentation.GetNthSegmentID(i) not in ["background", "background_scribbles", "foreground_scribbles"]
+            ]
+            currentItems = [self.ui.scribLabelComboBox.itemText(i) for i in range(self.ui.scribLabelComboBox.count)]
+            if len(segmentIds) == len(currentItems) and all([a == b for a, b in zip(segmentIds, currentItems)]):
```

Comment:
i think we don't need this.. we always populate the combo box and try to select previously selected label using parameterNode

Reply:
The main reason I took this approach is because we want to show a dialogue box to user whenever a different combobox label is selected for scribbles. We want to warn and ask for ok that the existing scribbles will be cleared. 
With the above logic for self.ui.labelComboBox the combobox was updated multiple times (clear, repopulate, select current text) - this isnt good for scribbles as then we will show multiple unnecessary dialogue boxes to user. 

Here I only update combobox for scribbles if there is a change in labels, otherwise I dont touch it.

Reply:
i meant something like this.. little less duplicate code.. 

```python
self.ui.labelComboBox.clear()
self.ui.scribLabelComboBox.clear()
if self._segmentNode:
    segmentation = self._segmentNode.GetSegmentation()
    totalSegments = segmentation.GetNumberOfSegments()
    segmentIds = [segmentation.GetNthSegmentID(i) for i in range(totalSegments)]
    for idx, segmentId in enumerate(segmentIds):
        segment = segmentation.GetSegment(segmentId)
        label = segment.GetName()
        if label not in ["foreground_scribbles", "background_scribbles"]:
            self.ui.labelComboBox.addItem(label)
        if label not in ["background", "foreground_scribbles", "background_scribbles"]:
            self.ui.scribLabelComboBox.addItem(label)
else:
    for label in self.info.get("labels", {}):
        self.ui.labelComboBox.addItem(label)
        if label != "background":
            self.ui.scribLabelComboBox.addItem(label)

currentLabel = self._parameterNode.GetParameter("CurrentLabel")
idx = self.ui.labelComboBox.findText(currentLabel) if currentLabel else 0
idx = 0 if idx < 0 < self.ui.labelComboBox.count else idx
self.ui.labelComboBox.setCurrentIndex(idx)

currentScribbleLabel = self._parameterNode.GetParameter("CurrentScribbleLabel")
idx = self.ui.scribLabelComboBox.findText(currentScribbleLabel) if currentScribbleLabel else 0
idx = 0 if idx < 0 < self.ui.scribLabelComboBox.count else idx
self.ui.scribLabelComboBox.setCurrentIndex(idx)

```

Reply:
I was trying to avoid having to change combobox label and call `onSelectScribLabel()` multiple times. This code makes changes to it multiple times, each time calling the method which might not be needed if labels in these comboboxes are not changing. 
For now, I have bypassed the notification message in `onSelectScribLabel()` by wrapping this piece of code within `self.ignoreScribblesLabelChangeEvent = True`. I am adding a commit with these changes and marking this as resolved.